### PR TITLE
Ideas to clarify documentation of common metadata

### PR DESCRIPTION
**Pull request by pp-mo**
**WIP**
Just to share some ideas.

I have now also pushed this as master-repo branch, so we can see [the built docs](https://scitools-docs.github.io/iris/common_metadata_docs/index.html).
***build coming shortly, I hope ? ...***

However, a lot of these are dev documentation, which is either private anyway, or possibly ought to be !

### CODE REVIEW

**Comment by pp-mo** [comment]
? not clear why this should be failing tests ?
Presumably, existing problems with the cube-arithmetic FB

File: lib/iris/common/metadata.py [hunk 1]
```diff
@@ -75,10 +82,33 @@ class BaseMetadata(metaclass=_NamedTupleMeta):
     """
     Container for common metadata.
 
+    This is the 'signature' of some cube component type (e.g. a cube or coord).
+    Being a 'namedtuple', it is an immutable, hashable value.
+
+    It's `_fields` property is a list of the 'signature' properties used by an
```
Comment (trexfeathers) [inline lib/iris/common/metadata.py:88]:
```suggestion
    Its `_fields` property is a list of the 'signature' properties used by an
```

File: lib/iris/common/metadata.py [hunk 2]
```diff
@@ -289,7 +323,16 @@ def MetadataManagerFactory(cls, **kwargs):
         fields will default to a value of 'None'.
 
     """
+    # Restrict factory to appropriate metadata classes only.
```
Comment (trexfeathers) [inline lib/iris/common/metadata.py:326]: Why is this repeating the same section already in `__init__`?

Comment (pp-mo) [reply lib/iris/common/metadata.py:326]: Good spot.
I removed the one in the __init__ : it is only ever called from the factory function itself, so let's just check once there.

**Pull request closed without merging**